_posts/2016-04-03-JabRef-and-JavaFX.md

```diff
@@ -0,0 +1,61 @@
+---
+title: JabRef and JavaFX
+id: jabref-javafx
+bg: orange
+color: white
+style: left
+---
+
+
+JabRef as of today is using swing components to build its gui.
+However as of java version 1.8 a new more promissing technology is available called JavaFX 8 which is a more advanced version of JavaFX 1.X or 2.X.
+Since JabRef recently switched to java version 1.8 it is getting more and more interesting how to migrate the user interface to this new technology.
+
+If we try to just include JavaFX components right now we would face a _IllegalStateException_ with the message _Toolkit not initialized_.
+The JavaFX toolkit gets automatically initialized when an application is started through the _launch_ method of a class that extends the _Application_ class from the package _javafx.application_.
+As JabRef has not been using any JavaFX components yet it does not start this way and we need to change its start behaviour.
+For this we have to look into the class _JabRefMain_ located in the package _net.sf.jabref_ of the source folder _src/main/java_.
+As mentioned earlier we need to inherit the _Application_ class which means we need to add `extends Application` after the class name.  
+Following is the new class signature.
+
+    public class JabRefMain extends Application {
```

Comment:
Please use [fenced code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/#syntax-highlighting) instead of intended blocks to enable language syntax highlighting. This should be working when the colors are fixed.